### DISCUSSION TITLE
Link from link/next is not working when deployed on dev

### DISCUSSION THREAD

**Discussion by deependras1110** [vercel/next.js#38349] [Help]
```
<Link href={href} >
            <a className={className}>
                     {children}
            </a>
</Link>
```
1) props are working fine
2) Before using link, used <a> with onClick, that has router.push(href), it's working fine in dev, staging, prod

What could be the issues? Tried some suggestions such as using as, prefetch={false} as props to Link

**Comment by icyJoseph** [comment]
Mm well, this is working fine for me. 

- Do you have more surrounding code? Perhaps a link to a repository?
- If you inspect the HTML, do you see an anchor tag with `<a href="` ?
- The right import name is `import Link from 'next/link';`, but I don't think this is the issue.
- What's the value of `href`?
- Is `children` perhaps stoping the click event from propagating? 
- Does it work with disabled JavaScript? 
- What next.js version are you running?

**Reply by deependras1110** [reply]
Really appreacite your response, here are the answers to some of your questions. 
Please NOTE: The router.push method is working fine in deployed version but need to change to Link. 
1) I understand but I wont be able to provide neither the repository nor the reproduction. 
2) In first place, links are not getting rendered, so when I inspect the HTML, I don't see the respective tags of the link. 
3) I have some other link's which have only <a>, those are gettig dispalyed in viewport and working fine. 
4) The value of href is an absolute path, like domain/page1, domain/page2
5) Checked the children, not stopping any click event from propogating
**6) So I even have issues with css, styling is not getting implemented but when I disabled the javascript,styling is working fine. What could this mean?** 
7) 12.1.0

**Reply by icyJoseph** [reply]
> Really appreacite your response, here are the answers to some of your questions. Please NOTE: The router.push method is working fine in deployed version but need to change to Link.
> 

That almost doesn't matter, if anything it makes the problem even weirder, because the Link is taking care of rendering proper HTML, but at the end of the day it invokes the `router.push` or `replace` methods. SPA routing is a bit of a façade. 

Anyhow...

> 1. I understand but I wont be able to provide neither the repository nor the reproduction.

At least more surrounding code is necessary. Do you see any logs? issues? HTML result snippets, component debug info?

To get component debug info, you can use the React Dev Tools, and find the component in the React Tree, click on it, and from the console do `$r` and hit enter, that'll log hooks and props at the time of hitting enter.

If the deployed version is available on the internet, then please share a link to it.

> 2. In first place, links are not getting rendered, so when I inspect the HTML, I don't see the respective tags of the link.
> 3. I have some other link's which have only , those are gettig dispalyed in viewport and working fine.

This is rather scary. This tells me that there's a lot of code context which is not being represented here. 

> 4. The value of href is an absolute path, like domain/page1, domain/page2
> 5. Checked the children, not stopping any click event from propogating

With the little information we have we could think that these are not the problems then

**6) So I even have issues with css, styling is not getting implemented but when I disabled the javascript,styling is working fine. What could this mean?**
> 6. 12.1.0

Again, rather difficult to know, let alone guess. I would think that there's user land code that, when JS is disabled, allows the CSS to go through fine, but that otherwise might be interfering with both navigation and styling.

> When JS is disabled, SPA transitions stop working, and good ol' document navigation kicks in

I wonder if you have some sort of name collision/shadowing, something else called Link that's declared on a lower scope.

**Reply by deependras1110** [reply]
Is there anything other than Link, and  router.push? Any alternative way to go, like  building a Link componet from scratch using Vanilla.js. Or maybe try to inject React-Router.

**Reply by icyJoseph** [reply]
That's a huge red flag right there. 

I have to speak honestly here. There're test, both unit and with actual browser instances that protect the link behavior. It doesn't just fail.

Start a new next.js project with create next app and see how it works there by itself.

As I have said before, in SPAs the Link takes care of informing the router about the transition, render correct HTML and then do the push or replace.

We are all trying to help from the very little code you have shared. Trying to re-invent the wheel or try to use duck tape with another library is not gonna work for SPA transitions. 

Are you perhaps using a custom component as children to Link? Those need ref forwarding to work, and the link needs something called passHref. 

I don't understand how you cannot make a stackblitz reproduction demo? What else is happening in this code base? I remember helping someone who was running a next.js inside a streaming platform and it had some implications, that broke other features. Is this perhaps your case?

If you want you can invite me as collaborator and I can take a quick look.

**Reply by deependras1110** [reply]
Appreciate your help! 
Not using custom componet as children to link!

**Comment by ryanhaticus** [comment]
Can you provide a reproduction repository and link it here?
That would help tremendously.

**Comment by deependras1110** [comment]
Hi,

I am sorry I won’t be able to share the repository.

Sincerely,
Deependra

**Reply by ryanhaticus** [reply]
To be honest, without a reproduction, this issue will be immensely difficult to pinpoint.

Consider creating another repository with a sample of code that is causing issues.
Without this, I'm afraid there isn't much we can do.
